sr_gui_change_controllers/src/sr_gui_change_controllers/change_controllers.py

```diff
@@ -51,64 +57,136 @@ def __init__(self, context):
+            self._widget.rh_traj.setIcon(self.CONTROLLER_OFF_ICON)
+            self._widget.rh_traj.toggled.connect(
+                self.teach_mode_button_toggled_rh)
+            self._rh_teach_buttons.append(self._widget.rh_traj)
+
+            self._widget.rh_pos.setIcon(self.CONTROLLER_OFF_ICON)
+            self._widget.rh_pos.toggled.connect(
+                self.teach_mode_button_toggled_rh)
+            self._rh_teach_buttons.append(self._widget.rh_pos)
+
+            self._widget.rh_teach.setIcon(self.CONTROLLER_OFF_ICON)
+            self._widget.rh_teach.toggled.connect(
+                self.teach_mode_button_toggled_rh)
+            self._rh_teach_buttons.append(self._widget.rh_teach)
+            # hide teach mode if arm...
+            if 'ra_' in self.avaliable_controller_groups:
+                self._widget.rh_teach.hide()
+
+        if 'lh_' not in self.avaliable_controller_groups:
+            self._widget.lh_group.setDisabled(True)
+        else:
+            self._widget.lh_traj.setIcon(self.CONTROLLER_OFF_ICON)
+            self._widget.lh_traj.toggled.connect(
+                self.teach_mode_button_toggled_lh)
+            self._lh_teach_buttons.append(self._widget.lh_traj)
+
+            self._widget.lh_pos.setIcon(self.CONTROLLER_OFF_ICON)
+            self._widget.lh_pos.toggled.connect(
+                self.teach_mode_button_toggled_lh)
+            self._lh_teach_buttons.append(self._widget.lh_pos)
+
+            self._widget.lh_teach.setIcon(self.CONTROLLER_OFF_ICON)
+            self._widget.lh_teach.toggled.connect(
+                self.teach_mode_button_toggled_lh)
+            self._lh_teach_buttons.append(self._widget.lh_teach)
+            # hide teach mode if arm...
+            if 'la_' in self.avaliable_controller_groups:
+                self._widget.lh_teach.hide()
+
+        # TODO: add buttons when more controller modes are available for arms.
+        self._widget.ra_group.hide()
+        self._widget.la_group.hide()
+
+        self._widget.information_box.clicked.connect(self.display_information)
+        self.confirm_current_control()
+
+    def find_active_control_groups(self):
+        joint_states = rospy.wait_for_message('/joint_states', JointState)
+        avaliable_robot_names = []
+        for robot_name in self.robot_names:
+            for joint in joint_states.name:
+                if joint.startswith(robot_name):
+                    if robot_name not in avaliable_robot_names:
+                        avaliable_robot_names.append(robot_name)
+        return avaliable_robot_names
+
+    def confirm_current_control(self):
+        list_controllers = rospy.ServiceProxy(
+            'controller_manager/list_controllers', ListControllers)
+        try:
+            resp1 = list_controllers()
+        except rospy.ServiceException:
+            rospy.logerr(
+                "Couldn't get list of controllers from controller_manager/list_controllers service")
+            return
```

Comment:
do you want the program to keep running if list controller is empty? what happens to the higher level? if you want the program to stop you should raise the exception